Lib/asyncore.py

```diff
@@ -144,17 +144,17 @@ def poll(timeout=0.0, map=None):
         r, w, e = select.select(r, w, e, timeout)
 
         for obj in r:
-            if obj.fileno() is None:
+            if obj.fileno() is -1:
```

Comment:
This may fail, there may be multiple -1 instances, unlike the single None instance. Better use == when comparing a number. But see my comment bellow about using obj.closing instead.